### PR TITLE
[#370] catch the exception

### PR DESCRIPTION
**Pull request by BeckerFrank**

### CODE REVIEW

File: mylyn.context/org.eclipse.mylyn.java.ui/src/org/eclipse/mylyn/internal/java/ui/editor/FocusedJavaAllProposalComputer.java [hunk 1]
```diff
@@ -39,8 +39,12 @@ public FocusedJavaAllProposalComputer() {
 	@Override
 	public List computeCompletionProposals(ContentAssistInvocationContext context, IProgressMonitor monitor) {
 		if (shouldReturnResults()) {
-			List proposals = super.computeCompletionProposals(context, monitor);
-			return FocusedJavaProposalProcessor.getDefault().projectInterestModel(this, proposals);
+			try {
+				List proposals = super.computeCompletionProposals(context, monitor);
+				return FocusedJavaProposalProcessor.getDefault().projectInterestModel(this, proposals);
+			} catch (Exception e) {
+				return Collections.emptyList();
```
Comment (ruspl-afed) [inline mylyn.context/org.eclipse.mylyn.java.ui/src/org/eclipse/mylyn/internal/java/ui/editor/FocusedJavaAllProposalComputer.java:46]: I don't know when this exception occurs. Should we just ignore it or log as well?

Comment (BeckerFrank) [reply mylyn.context/org.eclipse.mylyn.java.ui/src/org/eclipse/mylyn/internal/java/ui/editor/FocusedJavaAllProposalComputer.java:46]: > I don't know when this exception occurs. Should we just ignore it or log as well?

This exception can occur with invalide java code, please see #370.
Example NPE in org.eclipse.jdt.internal.compiler.ast.*

So I think that logging this type of error is of no use.

Comment (ruspl-afed) [reply mylyn.context/org.eclipse.mylyn.java.ui/src/org/eclipse/mylyn/internal/java/ui/editor/FocusedJavaAllProposalComputer.java:46]: Thank you for confirming @BeckerFrank

**Review by ruspl-afed — APPROVED** [review]
LGTM

**Pull request merged**